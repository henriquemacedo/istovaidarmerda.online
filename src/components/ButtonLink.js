import React from "react";
import PropTypes from "prop-types";
import Link from "next/link";
import styled from "styled-components";

const Button = styled.a`
	border: 3px solid ${(props) => (props.borderColor ? props.borderColor : "var(--darkBrown)")};
	border-radius: 999px;
	background-color: ${(props) => (props.bgColor ? props.bgColor : "var(--lightBrown)")};
	width: min-content;
	height: 84px;
	padding: 0 32px;
	display: flex;
	justify-content: center;
	align-items: center;
	font-weight: var(--fontBlack);
	text-transform: uppercase;
	color: var(--light);
	transition-duration: 0.3s;
	cursor: pointer;
`;

const ButtonLink = ({ borderColor, bgColor, url, value }) => (
	<Link href={url}>
		<Button borderColor={borderColor} bgColor={bgColor}>
			{value}
		</Button>
	</Link>
);

ButtonLink.defaultProps = {
	borderColor: "#4d352d",
	bgColor: "#946037",
};

ButtonLink.propTypes = {
	borderColor: PropTypes.string,
	bgColor: PropTypes.string,
	url: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
};

export default ButtonLink;
