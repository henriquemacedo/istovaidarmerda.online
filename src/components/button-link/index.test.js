import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ButtonLink from "./index";

describe("src/components/button-link", () => {
  /**
   * @type {IButtonLinkProps}
   */
  let props;

  beforeEach(() => {
    props = {
      borderColor: "#4d352d",
      bgColor: "#946037",
      external: false,
      url: "https://twitter.com",
      label: "Follow on Twitter",
      value: "Twitter",
    };
  });

  describe("render", () => {
    it("should render without errors", () => {
      const component = render(
        <ButtonLink
          borderColor={props.borderColor}
          bgColor={props.bgColor}
          external={props.external}
          url={props.url}
          label={props.label}
          value={props.value}
        />,
      );

      expect(component).toMatchSnapshot();
    });

    it("should render an icon", () => {
      render(
        <ButtonLink
          borderColor={props.borderColor}
          bgColor={props.bgColor}
          external={props.external}
          url={props.url}
          label={props.label}
          value={props.value}
          icon="twitter"
        />,
      );

      const icon = screen.getByRole("presentation");

      expect(icon).toBeVisible();
    });

    it("should render an external icon", () => {
      render(
        <ButtonLink
          borderColor={props.borderColor}
          bgColor={props.bgColor}
          url={props.url}
          label={props.label}
          value={props.value}
          external
        />,
      );

      const icon = screen.getByRole("link");

      expect(icon).toHaveAttribute("rel", "noopener noreferrer");
    });
  });
});
