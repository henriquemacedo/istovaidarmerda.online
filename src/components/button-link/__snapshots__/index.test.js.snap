// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/button-link render should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Follow on Twitter"
        class="styles__Button-e7r930-0 kwPxEs"
        data-testid="button-link"
        href="https://twitter.com"
        target="_self"
      >
        Twitter
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Follow on Twitter"
      class="styles__Button-e7r930-0 kwPxEs"
      data-testid="button-link"
      href="https://twitter.com"
      target="_self"
    >
      Twitter
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
